_posts/2022/2022-11-14-safely-remove-a-column-field-from-active-record.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Safely Remove a Column from an Active Record Model"
+description: "Avoid migration-related errors when you’re tidying your database"
+layout: article
+category: ruby
+image:
+  base: "2022/safely-remove-a-column-field-from-active-record"
+  alt: "A partial column against flat background"
+  credit: "ANIRUDH"
+  source: "https://unsplash.com/photos/jtiQCAAzLYY"
+---
+
+Adding and deploying new columns to an existing Active Record model is often straightforward. Typically the first deployment runs your migrations and then new code that uses the new database column is released afterward.
+
+However, when removing a column you can more easily generate problems. Active Record caches database columns when it spins up a Rails application. When the column is removed from the database table it causes exceptions until your app reboots or redeploys.
+
+It's worth a multi-step strategy when removing columns in your database.
+
+
+## Use...
+
+...this multi-step strategy.
+
+First tell Active Record to ignore the column. This stops the rest of your application being able to reference this column.
+
+```ruby
+class Thing < ApplicationRecord
+  # ...
+  self.ignored_columns = ["old_column"]
+  # ...
+end
+```
+
+You can use your test suite to find places you might be still using the column in your application. Once your tests are passing, deploy this code.
+
+In a seperate change you can create the migration to remove the actual database column.
+
+```ruby
+class RemoveOldColumnFromThings < ActiveRecord::Migration[7.0]
+  def change
+    remove_column :things, :old_column
+  end
+end
+```
+
+Then deploy, migrating the column into oblivion.
+
+Then in a final change, remove the `ignored_columns` line you added for the initial step and deploy once more.
```

Comment:
Lots of "Then..." used for each step...Maybe some can be "After that" or "Next"..?

Reply:
```suggestion
After that, in a final change, remove the `ignored_columns` line you added for the initial step and deploy once more.
```